framework/src/media/audio/audio_manager.c

```diff
@@ -140,121 +151,72 @@ static const struct audio_samprate_map_entry_s g_audio_samprate_entry[] = {
 /****************************************************************************
  * Public Functions
  ****************************************************************************/
-audio_manager_result_t find_input_audio_card(void)
+audio_manager_result_t find_audio_card(audio_card_type_t card_type)
 {
 	int ret = AUDIO_MANAGER_SUCCESS;
-	uint8_t card_id = -1;
-	uint16_t device_id = -1;
+	uint8_t card_id = INVALID_ID;
+	uint16_t device_id = INVALID_ID;
 	char type;
 	struct dirent *dir_entry;
 	DIR *dir_info;
 
+	audio_card_info_t *card = g_audio_out_cards;
+	int max_card_num = CONFIG_AUDIO_MAX_OUTPUT_CARD_NUM;
+	char type_chr = 'p';
+	int active_card_id = INVALID_ID;
+
 	if (!(dir_info = opendir("/dev/audio/"))) {
 		return AUDIO_MANAGER_DEVICE_FAIL;
 	}
 
 	while ((dir_entry = readdir(dir_info)) != NULL) {
 		// TODO: Add cases for various drivers. Currently, identify 'pcm' drivers only.
-		if ((dir_entry->d_name[0] != 'p') || (dir_entry->d_name[1] != 'c') || (dir_entry->d_name[2] != 'm') ||
-			(sscanf(&dir_entry->d_name[3], "C%uD%u%c", &card_id, &device_id, &type) != 3)) {
+		if ((dir_entry->d_name[0] != 'p') || (dir_entry->d_name[1] != 'c') || (dir_entry->d_name[2] != 'm') || (sscanf(&dir_entry->d_name[3], "C%uD%u%c", &card_id, &device_id, &type) != 3)) {
 			continue;
 		}
 
-		if ((type == 'c') && (card_id < CONFIG_AUDIO_MAX_INPUT_CARD_NUM) && (device_id < MAX_AUDIO_DEVICE_NUM)) {
-			snprintf(g_audio_in_cards[card_id].card_path, strlen("/dev/audio/") + strlen(dir_entry->d_name) + 1, "%s%s",
-					 "/dev/audio/", dir_entry->d_name);
-
-			g_actual_audio_in_card_id = card_id;
-			g_audio_in_cards[card_id].status = AUDIO_CARD_READY;
-			g_audio_in_cards[card_id].device_id = device_id;
-			g_audio_in_card_num++;
-
-			ret = get_supported_sample_rate(INPUT, g_audio_in_cards[card_id].card_path);
-			ASSERT(ret == AUDIO_MANAGER_SUCCESS);
-
-			medvdbg("Found an input audio card, id=%d, count=%d,  card_path=%s\n", card_id, g_audio_in_card_num,
-					g_audio_in_cards[card_id].card_path);
+		if (card_type == INPUT) {
```

Comment:
I think this line needs be changed into  "if (type != type_chr) {"

Reply:
I don't understand, but type_chr is changed line #180 if card_type is INPUT, initial value is 'c' (see Line #165)